app/models/delay_henka/scheduled_action.rb

```diff
@@ -0,0 +1,44 @@
+module DelayHenka
+  class ScheduledAction < ApplicationRecord
+
+    STATES = {
+      STAGED: 'staged',
+      COMPLETED: 'completed',
+    }
+
+    belongs_to :actionable, polymorphic: true
+
+    validates :submitted_by_id, presence: true
+    validates :schedule_at, presence: true
+    validates :state, inclusion: { in: STATES.values }
+    after_initialize :set_initial_state, if: :new_record?
+
+    scope :staged, -> { where(state: STATES[:STAGED]) }
+
+    def self.schedule(record:, method_name:, arguments:, by_id:, schedule_at: Time.current)
+      DelayHenka::ScheduledAction.create(
+        actionable: record,
+        method_name: method_name,
+        arguments: arguments,
+        submitted_by_id: by_id,
+        schedule_at: schedule_at
+      )
+    end
+
+    def apply_action
+      if actionable
+        update(state: STATES[:COMPLETED])
+        actionable.send(method_name, *arguments)
```

Comment:
@zinosama i thought during exec the method, exception might happen, so need set state in advance